enhancements/index-init.md

```diff
@@ -0,0 +1,64 @@
+---
+title: index-init
+authors:
+  - "@kevinrizza"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-08
+last-updated: 2020-05-08
+status: implementable
+---
+
+# index-init
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The purpose of this enhancement is to define a new function for `opm index` which would generate initial index images with empty databases. Today, this can technically be accomplished by using `opm index add -b ""`, but semantically that is very unintuitive. The intention of this enhancement is to define a separate command, `opm index init`, to bootstrap empty index images . 
+
+## Motivation
+
+The primary motivating driver for this is to give CI pipelines utilizing `opm` a way to differentiate between initializing an index image and actually adding bundles to that initial image. From a CI tool's perspective, validation and rules would reasonably expect that `--from-index` should always be set, which means that the initial bootstrapping process becomes necessary. However, rather than expect users to use `opm index add` in an unintuitive and semantically confusing way, we should instead enable this as a first class feature.
+
+### Goals
+
+* `opm index init` can generate fresh empty initial index images
```

Comment:
Is there a way that makes sense (semantically) that we can support swapping base images for existing indexes as well?

Reply:
Is the goal "have the exact same image except for a different base image"? Or is the goal "build a new image that has the same indexed content"?

If the answer is the latter, I think this could be as simple as adding `--from-index` to `opm index init`, which would just also include grabbing the index database instead of initializing it.

Reply:
The latter - "build a new image that has the same indexed content"